aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferUtility.java

```diff
@@ -597,6 +605,79 @@ public TransferObserver upload(String key, File file, ObjectMetadata metadata, C
         return upload(getDefaultBucketOrThrow(), key, file, metadata, cannedAcl, listener);
     }
 
+    /**
+     * Starts uploading the inputStream to the <b>default</b> bucket, using the given key.
+     *
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     */
+    public void uploadInputStream(String key, InputStream inputStream) {
+        try {
+            final File file = writeInputStreamToFile(inputStream);
+            TransferObserver observer = upload(key, file);
+
+            observer.setTransferListener(new TransferListener() {
+                @Override
+                public void onStateChanged(int id, TransferState state) {
+                    switch (state) {
+                        case COMPLETED:
+                            file.delete();
+                            break;
+                        case FAILED:
+                            throw new RuntimeException("Transfer failed.");
+                    }
+                }
+
+                @Override
+                public void onProgressChanged(int id, long bytesCurrent, long bytesTotal) {}
+
+                @Override
+                public void onError(int id, Exception ex) {
+                    throw new RuntimeException(ex);
+                }
+            });
+        } catch (IOException e) {
+            LOGGER.error("Failed writing the input stream into a file: ", e);
+        }
+    }
+
+    /**
+     * Starts uploading the inputStream to the given bucket, using the given key.
+     *
+     * @param bucket        The name of the bucket to upload the new object to.
+     * @param key           The key in the specified bucket by which to store the new object.
+     * @param inputStream   The input stream to upload.
+     */
+    public void uploadInputStream(String bucket, String key, InputStream inputStream) {
+        try {
+            final File file = writeInputStreamToFile(inputStream);
+            TransferObserver observer = upload(bucket, key, file);
+
+            observer.setTransferListener(new TransferListener() {
+                @Override
+                public void onStateChanged(int id, TransferState state) {
+                    switch (state) {
+                        case COMPLETED:
+                            file.delete();
+                            break;
+                        case FAILED:
+                            throw new RuntimeException("Transfer failed.");
+                    }
+                }
+
+                @Override
+                public void onProgressChanged(int id, long bytesCurrent, long bytesTotal) {}
+
+                @Override
+                public void onError(int id, Exception ex) {
+                    throw new RuntimeException(ex);
+                }
+            });
+        } catch (IOException e) {
+            LOGGER.error("Failed writing the input stream into a file: ", e);
```

Comment:
How will the user know that these errors have a occurred? We need a way to pass these back to the user.